### DISCUSSION TITLE
Docker Persistance Storage

### DISCUSSION THREAD

**Discussion by wuast94** [emqx/emqx#8296] [Q&A]
How to setup Docker for Persitance storage? 

I tryed to find something in the docs but i cant figure out what volumes have to be mapped to get persistant storage. 

also configs dont get created on start? i have to map a dummy folder, connect terminal, copy files needed and than remap.
/data:/dummy
cp files there and than chaneg mapping to
/data:/opt/emqx/needed_folders

**Comment by zmstone** [comment]
EMQX state is stored in `/opt/emqx/data`.
Note: It’s important to use static node name when persisting EMQX data dir.

**Reply by wuast94** [reply]
Is this the only place? etc contains config too or am I wrong? And are there any other dirs that contains persistent data?

**Reply by zmstone** [reply]
Yes data dir is the only directory to persist EMQX’s state.

**Reply by zmstone** [reply]
etc is ready only

**Reply by wuast94** [reply]
> etc is ready only

why are certs and acl.conf read only?

**Reply by zmstone** [reply]
They are read only configs to emqx.
Certs and acl are user’s provisions, if emqx doesn’t mutate it, it’s not its state, but user input.
Do you mean you want to persist its state including your changes in configuration?
 Then it could be anything
